src/main/java/it/pagopa/ecommerce/payment/methods/utils/UniqueIdUtils.java

```diff
@@ -1,19 +1,49 @@
 package it.pagopa.ecommerce.payment.methods.utils;
 
+import it.pagopa.ecommerce.payment.methods.exception.UniqueIdGenerationException;
+import it.pagopa.ecommerce.payment.methods.infrastructure.UniqueIdDocument;
+import it.pagopa.ecommerce.payment.methods.infrastructure.UniqueIdTemplateWrapper;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import reactor.core.publisher.Mono;
 
 import java.security.SecureRandom;
+import java.time.Duration;
 
 @Component
 public class UniqueIdUtils {
+    private final UniqueIdTemplateWrapper uniqueIdTemplateWrapper;
     private static final SecureRandom secureRandom = new SecureRandom();
     private static final String ALPHANUMERICS = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-._";
     private static final int MAX_LENGTH = 18;
+    private static final int MAX_NUMBER_ATTEMPTS = 3;
+    private static final String PRODUCT_PREFIX = "E";
```

Comment:
I would make these package-private so they can be used in tests

```suggestion
    static final int MAX_NUMBER_ATTEMPTS = 3;
    static final String PRODUCT_PREFIX = "E";
```